webapp/src/components/admin_settings/jitsi_settings.tsx

```diff
@@ -0,0 +1,243 @@
+import React, {useCallback, useEffect, useState, useMemo, ChangeEvent} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {AdminConfig} from 'mattermost-redux/types/config';
+
+import {id as pluginId} from 'manifest';
+import I18nProvider from 'components/i18n_provider';
+import JaaSSection from './jaas_section';
+import {RadioField} from '../RadioField';
+import JitsiSection, {JITSI_NAMING_SCHEME} from './jitsi_section';
+import {InputElementType} from 'types';
+
+type Props = {
+    id: string,
+    label: string,
+    value: Settings,
+    disabled: boolean,
+    config: AdminConfig,
+    license: object,
+    setByEnv: boolean,
+    onChange: Function,
+    registerSaveAction: Function,
+    setSaveNeeded: Function,
+    unRegisterSaveAction: Function
+};
+
+type Settings = {
+    usejaas?: boolean,
+    jitsiurl?: string,
+    jitsiembedded?: boolean,
+    jitsinamingscheme?: string,
+    jitsijwt?: boolean,
+    jitsiappid?: string,
+    jitsiappsecret?: string,
+    jitsilinkvalidtime?: number,
+    jitsicompatibilitymode?: boolean,
+    jaasappid?: string,
+    jaasapikey?: string,
+    jaasprivatekey?: string
+};
+
+const JITSI_MODE = 'JITSI_MODE';
+const JAAS_MODE = 'JAAS_MODE';
+
+const DEFAULT_JITSI_URL = 'https://meet.jit.si';
+const JITSI_LINK_VALID_TIME = 30;
+const DEFAULT_SETTINGS: Settings = {
+    usejaas: false,
+    jitsiurl: DEFAULT_JITSI_URL,
+    jitsiembedded: false,
+    jitsinamingscheme: JITSI_NAMING_SCHEME.WORDS,
+    jitsijwt: false,
+    jitsiappid: '',
+    jitsiappsecret: '',
+    jitsilinkvalidtime: JITSI_LINK_VALID_TIME,
+    jitsicompatibilitymode: false,
+    jaasappid: '',
+    jaasapikey: '',
+    jaasprivatekey: ''
+};
+
+const JitsiSettings = ({id, value, disabled, config, onChange, setSaveNeeded}: Props) => {
+    const selectedMode = value?.usejaas ? JAAS_MODE : JITSI_MODE;
+
+    const selectedSettings = useMemo(() => ({
+        jitsiurl: config.PluginSettings.Plugins[pluginId].jitsiurl ?? DEFAULT_SETTINGS.jitsiurl,
```

Comment:
To clean up some code duplication

```suggestion
    const config = config.PluginSettings.Plugins[pluginId];

    const selectedSettings = useMemo(() => ({
        jitsiurl: config.jitsiurl ?? DEFAULT_SETTINGS.jitsiurl,
```